inference-engine/src/mkldnn_plugin/nodes/mkldnn_roi_pooling_node.cpp

```diff
@@ -409,159 +404,279 @@ void MKLDNNROIPoolingNode::initSupportedPrimitiveDescriptors() {
-    if (!selectedPrimitiveDescriptor)
+    auto selectedPD = getSelectedPrimitiveDescriptor();
+    if (!selectedPD)
         IE_THROW() << "CPU ROI Pooling node with name '" << getName() << "' doesn't have primitive descriptors.";
-    auto config = selectedPrimitiveDescriptor->getConfig();
-
-    const int simd_w = mayiuse(cpu::x64::avx512_common) ? 16 : 8;
-    jpp.c_block = simd_w;
-
-    auto inDims = getParentEdgeAt(0)->getMemory().getStaticDims();
-    auto outDims = getChildEdgesAtPort(0)[0]->getMemory().getStaticDims();
-
-    jpp.mb = outDims[0];
-    jpp.c = rnd_up(inDims[1], simd_w);
-    jpp.ih = inDims[2];
-    jpp.iw = inDims[3];
-    jpp.oh = outDims[2];
-    jpp.ow = outDims[3];
 
-    jpp.spatial_scale = spatial_scale;
-    jpp.pooled_h = pooled_h;
-    jpp.pooled_w = pooled_w;
+    refParams.c_block = mayiuse(cpu::x64::avx512_common) ? 16 : 8;;
+    refParams.nb_c_blocking = mayiuse(cpu::x64::avx512_common) ? 15 : 7;
+    refParams.alg = getAlgorithm();
 
-    jpp.nb_c = jpp.c / jpp.c_block;
+    const auto& config = selectedPD->getConfig();
+    refParams.src_prc = config.inConfs[0].desc->getPrecision();
+    refParams.dst_prc = config.outConfs[0].desc->getPrecision();
+    refParams.src_data_size = refParams.src_prc.size();
+    refParams.dst_data_size = refParams.dst_prc.size();
 
-    jpp.nb_c_blocking = mayiuse(cpu::x64::avx512_common) ? 15 : 7;
+    if (inputShapesDefined()) {
+        if (needPrepareParams())
+            prepareParams();
+        updateLastInputDims();
+    }
+}
 
-    auto selectedPD = getSelectedPrimitiveDescriptor();
-    jpp.src_prc = selectedPD->getConfig().inConfs[0].desc->getPrecision();
-    jpp.dst_prc = selectedPD->getConfig().outConfs[0].desc->getPrecision();
-    jpp.src_data_size = jpp.src_prc.size();
-    jpp.dst_data_size = jpp.dst_prc.size();
+void MKLDNNROIPoolingNode::execute(mkldnn::stream strm) {
+    if (execPtr) {
+        auto &srcMemory0 = getParentEdgeAt(0)->getMemory();
+        auto &srcMemory1 = getParentEdgeAt(1)->getMemory();
+        auto &dstMemory = getChildEdgeAt(0)->getMemory();
 
-    jpp.alg = getAlgorithm();
+        auto src_strides = srcMemory0.GetDescWithType<BlockedMemoryDesc>()->getStrides();
+        auto dst_strides = dstMemory.GetDescWithType<BlockedMemoryDesc>()->getStrides();
+        size_t src_roi_step = srcMemory1.GetDescWithType<BlockedMemoryDesc>()->getStrides()[0];
 
-    if (mayiuse(cpu::x64::avx512_common)) {
-        roi_pooling_kernel.reset(new jit_uni_roi_pooling_kernel_f32<cpu::x64::avx512_common>(jpp));
-    } else if (mayiuse(cpu::x64::avx2)) {
-        roi_pooling_kernel.reset(new jit_uni_roi_pooling_kernel_f32<cpu::x64::avx2>(jpp));
-    } else if (mayiuse(cpu::x64::sse41)) {
-        roi_pooling_kernel.reset(new jit_uni_roi_pooling_kernel_f32<cpu::x64::sse41>(jpp));
+        execPtr->exec(srcMemory0.GetPtr(), srcMemory1.GetPtr(), dstMemory.GetPtr(), src_strides, dst_strides, src_roi_step);
```

Comment:
As for me it is much better just to pass srcMemory0, srcMemory1, dstMemory as exec() params and extract all necessera information inside.